src/app/model/ChessEngine.ts

```diff
@@ -68,7 +70,78 @@ export class ChessEngine implements ChessModel {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
 
+  private returnWirtualProposedBoardAfterMove(chessboard: Chessboard, squareFrom: Square, squareTo: Square): Chessboard {
+    const wirtualPorposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = wirtualPorposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    return wirtualPorposedChessboard;
+  }
+
+  public static kingPosition(chessboard: Chessboard, kingSide: Side): Square | undefined {
```

Comment:
Typ tej metody nam troche klamie. To tak jakby moglo nie byc Krola na planszy. A wg. zasad szachow MUSI byc :) Wiec nie powinno byc mozliwosci zwrocenia undefined.

Reply:
Hmmm... rozumiem - testy bez Króla w ulozeniu szachownicy. Kurde - racja :/ To wczesniej zrobilismy blad :P Ale to trzeba juz isc w to dalej - zeby nie zepsuc za duzo, a nawjyzej w osobnym PR zmienic.